modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/DefaultCommerceOrderValidatorImpl.java

```diff
@@ -65,19 +74,48 @@ public CommerceOrderValidatorResult validate(
 					locale, "this-order-has-already-been-checked-out", null));
 		}
 
-		CPDefinitionInventory cpDefinitionInventory =
-			_cpDefinitionInventoryLocalService.
-				fetchCPDefinitionInventoryByCPDefinitionId(
-					cpInstance.getCPDefinitionId());
-
-		CPDefinitionInventoryEngine cpDefinitionInventoryEngine =
-			_cpDefinitionInventoryEngineRegistry.getCPDefinitionInventoryEngine(
-				cpDefinitionInventory);
+		CPDefinitionInventoryEngine cpDefinitionInventoryEngine = null;
+		long cpConfigurationListId = 0;
+
+		if (FeatureFlagManagerUtil.isEnabled("LPD-10889")) {
+			CommerceChannel commerceChannel =
+				_commerceChannelLocalService.getCommerceChannelByGroupId(
+					commerceOrder.getGroupId());
+
+			CPConfigurationList cpConfigurationList =
+				_cpConfigurationListDiscovery.getCPConfigurationList(
+					cpInstance.getCompanyId(), cpInstance.getGroupId(),
+					commerceOrder.getCommerceAccountId(),
+					commerceChannel.getCommerceChannelId(),
+					commerceOrder.getCommerceOrderTypeId());
+
+			cpConfigurationListId =
+				cpConfigurationList.getCPConfigurationListId();
+
+			CPConfigurationEntry cpConfigurationEntry =
```

Comment:
can this be null? if yes we will have a NPE, if not why the fetch instead of the get?

Reply:
no, I've added a check that if the list is empty, it looks for the master configuration list